docs/deployment-integration.md

```diff
@@ -0,0 +1,176 @@
+
+> **Note**: If a network uses both Permissioned CosmWasm and 
+> Permissioned BSN Registration, two separate governance proposals 
+are currently required. A unified
+> governance flow requiring only a single governance proposal is being
+> examined.
+
+For the rest of the document, we will assume a Permissionless CosmWasm and BSN
+Registration network for simplicity. Listings that would require governance
+will be highlighted appropriately.
+
+## Instantiation
+<img width="3033" height="275" alt="instantiate" src="./assets/instantiate.png" />
+
+The Rollup BSN contract is instantiated as follows:
+```shell
+INSTANT_MSG=$(cat <<EOF
+{
+  "admin": "$ADMIN_ADDRESS",
+  "bsn_id": "$BSN_ID",
+  "max_msgs_per_interval": $MAX_MSGS_PER_INTERVAL,
+  "min_pub_rand": $MIN_PUB_RAND,
+  "rate_limiting_interval": $RATE_LIMITING_INTERVAL,
+  "allowed_finality_providers": ["$FP_PUBKEY_1", "$FP_PUBKEY_2"]
+}
+EOF
+)
+
+babylond tx wasm instantiate $CODE_ID '$INSTANT_MSG'
+```
+
+The parameters:
+* `admin`: The Babylon Genesis address of the contract administrator
+* `bsn_id`: The unique identifier for the BSN rollup
+  Ensure that this is unique and not already used by another BSN on Babylon
+  Genesis as this will affect your registration
+* `min_pub_rand`: Minimum public randomness values included per public
+  randomness commit by finality providers
+* `rate_limiting_interval`: Length (in blocks) of the interval for
+  rate-limiting submissions
+* `max_msgs_per_interval`: Maximum number of submissions allowed per finality
+  provider within each interval
+* `allowed_finality_providers`: List of finality providers from which the contract will
+  accept finality signatures and randomness
+
+<!-- TODO: once we have docs for the allow list and rate limiting, we can link
+those here -->
+
+## Rollup BSN Registration
+<img width="3032" height="287" alt="register" src="./assets/register.png" />
+
+After deploying and instantiating the Rollup BSN contract, the
+rollup must register on Babylon Genesis. Registration requires 
+submitting metadata to identify and describe the BSN
+
+To register the BSN, use:
+```shell
+babylond tx btcstkconsumer register-consumer \
+                                <consumer_id> \
+                                <consumer_name> \
+                                <consumer_ description> \
+                                <max_multi_staked_fps> \
+                                <rollup_finality_contract_address>
+```
+
+Required metadata for BSN registration:
+* `consumer_id`: BSN ID of your rollup (same as the BSN ID you used on contract
+  instantiation)
+* `consumer_name`: Human-readable name of your rollup BSN (e.g., `"DeFi Rollup Chain"`)
+* `consumer_description`: Brief description of the rollup BSN's purpose
+* `max_multi_staked_fps`: Maximum number of rollup FPs per BTC delegation. Prevents delegators from splitting
+ their BTC across too many networks (recommended: `3-5`)
```

Comment:
I think this feature has been removed from Babylon. Now Babylon enforces a global `max_multi_staked_fps` for all BSNs https://github.com/babylonlabs-io/babylon/blob/main/proto/babylon/btcstaking/v1/params.proto#L71-L73